authors/apps/comments/views.py

```diff
@@ -0,0 +1,156 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import generics, status
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from .serializers import CommentSerializer, ReplySerializer
+from ..profiles.models import Profile
+from .models import Comment, Reply
+from ..articles.models import Article
+from rest_framework.response import Response
+
+
+class CommentsAPIView(generics.ListCreateAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentSerializer
+    renderer_classes = (CommentRenderer,)
+
+    def get(self, request, slug):
+        article = get_object_or_404(Article, slug=slug)
+        comments = Comment.objects.filter(article=article)
+        serialize_data = self.serializer_class(comments, many=True)
+        return Response({"comments": serialize_data.data}, 
+                        status=status.HTTP_200_OK)
+
+    def post(self, request, slug):
+        article = get_object_or_404(Article, slug=slug)
+        data = request.data
+        author = Profile.objects.get(user=request.user)
+        serializer = self.serializer_class(data=data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(
+            author=author,
+            article=article
+        )
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class CommentDetailsAPIView(generics.RetrieveUpdateDestroyAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentSerializer
+    renderer_classes = (CommentRenderer,)
+
+    def required_objects(self, request, slug, comment_pk):
+
+        self.comment = get_object_or_404(Comment, pk=comment_pk)
+        self.data = request.data
+
+    def author_requirements(self, request):
+        self.requester = Profile.objects.get(user=request.user)
+        self.valid_author = self.comment.author == self.requester
+
+    def get(self, request, slug, comment_pk):
+        """gets one comment and its details"""
+        comment = get_object_or_404(Comment, pk=comment_pk)
+        serialize_data = self.serializer_class(comment)
+        return Response(serialize_data.data, status=status.HTTP_200_OK)
+
+    def patch(self, request, slug, comment_pk):
+        self.required_objects(request, slug, comment_pk)
+        self.author_requirements(request)
+
+        if not self.valid_author:
+            return Response(
+                {"response": "you can only update comments you made"},
+                status=status.HTTP_403_FORBIDDEN)
+
+        serializer = self.serializer_class(
+            self.comment, self.data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def delete(self, request, slug, comment_pk):
+        self.required_objects(request, slug, comment_pk)
+        self.author_requirements(request)
+
+        if not self.valid_author:
```

Comment:
@sebbss , raise a permission denied error for this if clause
`from rest_framework.exceptions import PermissionDenied`.
Look @Rhytah  profile task for how it can be done